provisioning_templates/snippet/redhat_register.erb

```diff
@@ -142,18 +142,18 @@ name: redhat_register
       subscription-manager config --server.proxy_user="<%= host_param('http-proxy-user') %>"
     <% end %>
     <% if host_param('http-proxy-password') %>
-      subscription-manager config --server.proxy_password="<%= host_param('http-proxy-password') %>"
+      subscription-manager config --server.proxy_password='<%= host_param("http-proxy-password") %>'
     <% end %>
     <% if host_param('http-proxy-port') %>
       subscription-manager config --server.proxy_port="<%= host_param('http-proxy-port') %>"
     <% end %>
   <% end %>
   <% if host_param('subscription_manager_username') && host_param('subscription_manager_password') %>
     <% if host_param('subscription_manager_pool') %>
-      subscription-manager register --name="<%= @host.name %>" --username="<%= host_param('subscription_manager_username') %>" --password="<%= host_param('subscription_manager_password') %>"
+      subscription-manager register --name="<%= @host.name %>" --username="<%= host_param('subscription_manager_username') %>" --password='<%= host_param("subscription_manager_password") %>'
       subscription-manager attach --pool="<%= host_param('subscription_manager_pool') %>"
     <% else %>
-      subscription-manager register --name="<%= @host.name %>" --username="<%= host_param('subscription_manager_username') %>" --password="<%= host_param('subscription_manager_password') %>" --auto-attach
+      subscription-manager register --name="<%= @host.name %>" --username="<%= host_param('subscription_manager_username') %>" --password='<%= host_param("subscription_manager_password") %>' --auto-attach
```

Comment:
Looks good to me but if you're touching this, would you mind also fixing name and username? I guess there are more strict validations for these, but still...